src/Sylius/Bundle/CoreBundle/Sitemap/Provider/ProductUrlProvider.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+ 
+namespace Sylius\Bundle\CoreBundle\Sitemap\Provider;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Sylius\Bundle\CoreBundle\Sitemap\Factory\SitemapUrlFactoryInterface;
+use Sylius\Component\Core\Model\Product;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class ProductUrlProvider implements UrlProviderInterface
+{
+    /**
+     * @var RepositoryInterface
+     */
+    private $repository;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var SitemapUrlFactoryInterface
+     */
+    private $sitemapUrlFactory;
+
+    /**
+     * @var Collection
+     */
+    private $urls;
+
+    /**
+     * @param RepositoryInterface $repository
+     * @param RouterInterface $router
+     * @param SitemapUrlFactoryInterface $sitemapUrlFactory
+     */
+    public function __construct(
+        RepositoryInterface $repository,
+        RouterInterface $router,
+        SitemapUrlFactoryInterface $sitemapUrlFactory
+    ) {
+        $this->repository = $repository;
+        $this->router = $router;
+        $this->sitemapUrlFactory = $sitemapUrlFactory;
+
+        $this->urls = array();
```

Comment:
If it's ``Collection`` as it is written in docblock, you should initialize it with i.e. ``new ArrayCollection()``.